sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/BasicAuthJcsmpSessionService.java

```diff
@@ -119,8 +116,9 @@ public MessageReceiver getReceiver() {
       this.messageReceiver =
           retryCallableManager.retryCallable(
               this::createFlowReceiver, ImmutableSet.of(JCSMPException.class));
+      this.messageReceiver.start();
     }
-    return this.messageReceiver;
+    return checkStateNotNull(this.messageReceiver);
```

Comment:
This check should not be necessary. Either `this.messageReceiver` was non-null or the body of the if statement ensures `this.messageReceiver` is non-null when the method returns.